packages/store/src/action.ts

```diff
@@ -13,15 +23,18 @@ export function Action(actions: any | any[], options?: ActionOptions) {
     }
 
     for (const action of actions) {
-      const type = getTypeFromKlass(action);
-      if (!meta.actions[type]) {
-        meta.actions[type] = [];
+      if (!action.type) {
+        action.type = new ActionToken(action.name);
+      }
+
+      if (!meta.actions[action.type]) {
```

Comment:
How does this `action.type` not represent a object here?

Reply:
it does represent an object. but when you do use an object as a key it calls the toString method and uses that as the actual key

Reply:
it works the same way as angular InjectionToken

Reply:
Oh didn't realize it did that. I figured it would just stick the object there.

Reply:
yeah I learned that as well.

Reply:
if you take the toString method off it breaks